From 832bb5a25776199f013a50b5bff7a5d6e8b15aa8 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 14 Dec 2011 14:12:38 +0800
Subject: [PATCH 249/505] ENGR00170168 ipudev unit test: add non
 FB_VMODE_YWRAP support

when use pan display, the case could be:
1. a small window wrap in a big frame buffer
2. a frame switch in a serial buffers

the ipuv3 fb driver used to support case 1, and for case 2,
if the fb format is interleaved, there is no problem, but for
non-interleaved format (like I420), there will be a display bug.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 test/mxc_ipudev_test/mxc_ipudev_test.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/mxc_ipudev_test/mxc_ipudev_test.c b/test/mxc_ipudev_test/mxc_ipudev_test.c
index 2486da5..67033c4 100644
--- a/test/mxc_ipudev_test/mxc_ipudev_test.c
+++ b/test/mxc_ipudev_test/mxc_ipudev_test.c
@@ -434,6 +434,7 @@ task_begin:
 		}
 
 		if (test_handle.show_to_fb) {
+			fb_var.vmode &= ~FB_VMODE_YWRAP;
 			fb_var.yoffset = (done_cnt % FB_BUFS) * fb_var.yres;
 			ret = ioctl(fd_fb, FBIOPAN_DISPLAY, &fb_var);
 			if (ret < 0) {
-- 
1.8.0

